### PR TITLE
Fs.verify.concurrency

### PR DESCRIPTION
**Pull request by kmlebedev**
# What problem are we solving?

Verifying is too slow, which makes it impossible to verify all files in a week during intensive writing

# How are we solving the problem?

Verify in parallel on each volume server

# How is the PR tested?

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/shell/command_fs_verify.go [hunk 1]
```diff
@@ -49,6 +54,7 @@ func (c *commandFsVerify) Do(args []string, commandEnv *CommandEnv, writer io.Wr
 	fsVerifyCommand := flag.NewFlagSet(c.Name(), flag.ContinueOnError)
 	c.verbose = fsVerifyCommand.Bool("v", false, "print out each processed files")
 	modifyTimeAgo := fsVerifyCommand.Duration("modifyTimeAgo", 0, "only include files after this modify time to verify")
+	c.parallelLimitByVolumeServer = fsVerifyCommand.Int("parallelLimitByVolumeServer", 0, "number of parallel verification per volume server")
```
Comment (chrislusf) [inline weed/shell/command_fs_verify.go]: just use "concurrency"?

Comment (kmlebedev) [reply weed/shell/command_fs_verify.go]: ok

**Comment by kmlebedev** [comment]
Tested

```
total 64729 directories, 18731395 files
verified 10359 files, error 0 files
```

File: docker/README.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ make
 
 list
 ```
-s3cmd --no-ssl --host=127.0.0.1:8333 ls s3://
+s3cmd --no-ssl --host=127.0.0.1:8333 --host-bucket=127.0.0.1:8333 ls s3://
```
Comment (chrislusf) [inline docker/README.md]: unrelated?

File: weed/shell/command_fs_verify.go [hunk 3]
```diff
@@ -84,15 +99,19 @@ func (c *commandFsVerify) collectVolumeIds() error {
 	eachDataNode(topologyInfo, func(dc string, rack RackId, nodeInfo *master_pb.DataNodeInfo) {
 		for _, diskInfo := range nodeInfo.DiskInfos {
 			for _, vi := range diskInfo.VolumeInfos {
-				c.volumeIds[vi.Id] = append(c.volumeIds[vi.Id], pb.NewServerAddressFromDataNode(nodeInfo))
+				volumeServer := pb.NewServerAddressFromDataNode(nodeInfo)
+				c.volumeIds[vi.Id] = append(c.volumeIds[vi.Id], volumeServer)
```
Comment (chrislusf) [inline weed/shell/command_fs_verify.go:103]: need to reset c.volumeIds before each run.

Same for other variables.

Very likely some other similar shell commands also have similar problem of unclean variables from previous runs.

Comment (kmlebedev) [reply weed/shell/command_fs_verify.go:103]: ok

File: docker/README.md (hunk 2 above)
Comment (kmlebedev) [reply docker/README.md]: yes

File: weed/shell/command_fs_verify.go [hunk 4]
```diff
@@ -67,12 +66,14 @@ func (c *commandFsVerify) Do(args []string, commandEnv *CommandEnv, writer io.Wr
 	}
 
 	c.modifyTimeAgoAtSec = int64(modifyTimeAgo.Seconds())
+	c.volumeIds = make(map[uint32][]pb.ServerAddress)
+	c.waitChan = make(map[string]chan struct{})
+	c.volumeServers = []pb.ServerAddress{}
```
Comment (chrislusf) [inline weed/shell/command_fs_verify.go:72]: we can add a defer clause to release these temp variables

**Comment by zemul** [comment]
Would you consider adding a deletion parameter?
